messaginginapp/src/main/java/io/customer/messaginginapp/ui/InlineInAppMessageView.kt

```diff
@@ -0,0 +1,89 @@
+package io.customer.messaginginapp.ui
+
+import android.content.Context
+import android.graphics.Color
+import android.util.AttributeSet
+import android.view.Gravity
+import android.widget.ProgressBar
+import androidx.annotation.AttrRes
+import androidx.annotation.ColorInt
+import androidx.annotation.StyleRes
+import androidx.core.graphics.drawable.DrawableCompat
+import io.customer.messaginginapp.R
+
+/**
+ * InlineInAppMessageView is a custom view that displays an inline in-app message for given
+ * elementId. The view shows a progress indicator while the message is loading and hides it once the
+ * message is displayed. If there is no message to display, the view will hide itself and display
+ * automatically when a new message is available.
+ *
+ * Example usage in XML:
+ *
+ * <io.customer.messaginginapp.ui.InlineInAppMessageView
+ *     android:id="@+id/inline_in_app_message"
+ *     android:layout_width="match_parent"
+ *     android:layout_height="wrap_content"
+ *     app:elementId="banner-message"/>
+ */
+class InlineInAppMessageView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    @AttrRes defStyleAttr: Int = 0,
+    @StyleRes defStyleRes: Int = 0
+) : InlineInAppMessageBaseView(context, attrs, defStyleAttr, defStyleRes),
+    InlineInAppMessageViewListener {
+    private val progressIndicator: ProgressBar = ProgressBar(context)
+
+    init {
+        var progressColor = resolveThemeColor(android.R.attr.colorControlActivated, Color.GRAY)
+        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.InlineInAppMessageView)
+        try {
+            elementId = typedArray.getString(R.styleable.InlineInAppMessageView_elementId)
+            if (typedArray.hasValue(R.styleable.InlineInAppMessageView_progressTint)) {
+                progressColor = typedArray.getColor(
+                    R.styleable.InlineInAppMessageView_progressTint,
+                    progressColor
+                )
+            }
+        } finally {
+            typedArray.recycle()
+        }
+        setupProgressIndicator(progressColor)
+        viewListener = this
+    }
+
+    private fun setupProgressIndicator(@ColorInt progressColor: Int) {
+        progressIndicator.isIndeterminate = true
+        progressIndicator.layoutParams = LayoutParams(
+            LayoutParams.WRAP_CONTENT,
+            LayoutParams.WRAP_CONTENT,
+            Gravity.CENTER
+        )
+        runCatching {
```

Comment:
do we need this? can this be simplified?

Reply:
Looking at Android docs, we don't really need `runCatching` here. But added it just to be safe when dealing with drawables since it doesn't do much harm even if it fails 😅